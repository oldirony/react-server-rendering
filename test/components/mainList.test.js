import React from 'react';
import MainList from '../../src/components/main-list/main-list';

const props = {
	ideas: {}
}

const ideas = {
	1 : {
		title: "title",
		isCompleted: false,
		timestamp: '2017-06-06T12:35:47+00:00'
	},
	2 : {
		title: "title",
		isCompleted: false,
		timestamp: '2017-06-06T12:35:47+00:00'
	}
}

test('It renders correctly without ideas', ()=>{
	let list = shallow(<MainList {...props}/>);
	expect(list).toMatchSnapshot();
})

test('It renders correctly without ideas with hidden completed ideas', ()=> {
	let list = shallow(<MainList {...props} hideCompletedIdeas={true}/>);
	expect(list).toMatchSnapshot();
})

test('It renders correctly with ideas', ()=> {
	let list = shallow(<MainList {...props} ideas={ideas}/>);
	expect(list).toMatchSnapshot();
})

test('It renders correctly with ideas with hidden completed ideas', ()=> {
	let list = shallow(<MainList {...props} hideCompletedIdeas={true} ideas={ideas}/>);
	expect(list).toMatchSnapshot();
})