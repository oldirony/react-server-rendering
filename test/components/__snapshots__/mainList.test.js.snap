// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly with ideas 1`] = `
<ul
  className="list"
>
  <IdeaListItem
    id="1"
    isCompleted={false}
    timestamp="2017-06-06T12:35:47+00:00"
    title="title"
  />
  <IdeaListItem
    id="2"
    isCompleted={false}
    timestamp="2017-06-06T12:35:47+00:00"
    title="title"
  />
</ul>
`;

exports[`It renders correctly with ideas with hidden completed ideas 1`] = `
<ul
  className="list"
>
  <IdeaListItem
    id="1"
    isCompleted={false}
    timestamp="2017-06-06T12:35:47+00:00"
    title="title"
  />
  <IdeaListItem
    id="2"
    isCompleted={false}
    timestamp="2017-06-06T12:35:47+00:00"
    title="title"
  />
</ul>
`;

exports[`It renders correctly without ideas 1`] = `
<div
  className="info centralColumn"
>
  <div
    className="content"
  >
    Waiting for you first idea!
  </div>
  <Link
    className="button"
    replace={false}
    to="/ideas/new"
  >
    Add new
  </Link>
</div>
`;

exports[`It renders correctly without ideas with hidden completed ideas 1`] = `
<div
  className="info centralColumn"
>
  <div
    className="content"
  >
    Well done, no uncompleted ideas!
  </div>
  <Link
    className="button"
    replace={false}
    to="/ideas/new"
  >
    Add new
  </Link>
</div>
`;
